File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int n = nums.length-1;
        if(nums.length==1) return nums[0];
        int s = 0 , e = n;
        while(s<e){
            int m = s+(e-s)/2;
            if(nums[m]>nums[n]){
                s = m+1;
            }
            else if(nums[m]<=nums[n]){
                e = m;
            }

        }
        return nums[e];
    }
}

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int low = 0, high = nums.size() -1 ;
        int minNum = INT_MAX;

        while(low<=high){
            int mid = low + ((high-low)/2);
            minNum = min(minNum , nums[mid]);
            if(nums[mid]<=nums[high]){
                high = mid-1;
            }
            else{
                low = mid+1;
            }
        }
        return minNum;
    }
};